pkgs/development/python-modules/catboost/nix-support.patch

```diff
@@ -0,0 +1,124 @@
+diff --git a/catboost/python-package/setup.py b/catboost/python-package/setup.py
+index 17f1d8ff14..26fef55b92 100644
+--- a/catboost/python-package/setup.py
++++ b/catboost/python-package/setup.py
+@@ -222,11 +222,12 @@ class build_ext(_build_ext):
+ 
+     def build_with_make(self, topsrc_dir, build_dir, catboost_ext, put_dir, verbose, dry_run):
+         logging.info('Buildling {} with gnu make'.format(catboost_ext))
+-        makefile = 'python{}.{}CLANG50-LINUX-X86_64.makefile'.format(python_version()[0], 'CUDA.' if self.with_cuda else '')
++        makefile = 'python{}.{}CLANG7-LINUX-X86_64.makefile'.format(python_version()[0], 'CUDA.' if self.with_cuda else '')
+         make_cmd = [
+             'make', '-f', '../../make/' + makefile,
+-            'CC=clang-5.0',
+-            'CXX=clang++-5.0',
++            'CC=clang',
++            'CXX=clang++',
++            'PYTHON=python2',
+             'BUILD_ROOT=' + build_dir,
+             'SOURCE_ROOT=' + topsrc_dir,
+         ]
+diff --git a/make/python2.CLANG7-LINUX-X86_64.makefile b/make/python2.CLANG7-LINUX-X86_64.makefile
+index e54b7078e8..0d230cdf0b 100644
+--- a/make/python2.CLANG7-LINUX-X86_64.makefile
++++ b/make/python2.CLANG7-LINUX-X86_64.makefile
+@@ -12,8 +12,8 @@ endef
+ _CC_VERSION = $(shell echo '$(_CC_TEST)' | $(CC) -E -P -)
+ $(info _CC_VERSION = '$(_CC_VERSION)')
+ 
+-ifneq '$(_CC_VERSION)' '7 0'
+-    $(error clang 7.0 is required)
++ifneq '$(_CC_VERSION)' '7 1'
++    $(error clang 7.1 is required)
```

Comment:
Why bump the clang version?

Reply:
We have clang 7.1.0 in 20.03 and 20.09. I think upstream states 7.0, because it's in Ubuntu 20.04.

Reply:
I would be fine with removing the check but bumping it seems a little to much maintenance work but should work.

Reply:
Checks removed.